core/integration/module_config_test.go

```diff
@@ -524,6 +524,240 @@ func (ConfigSuite) TestDaggerConfig(ctx context.Context, t *testctx.T) {
 	}
 }
 
+func (ConfigSuite) TestSDKConfig(ctx context.Context, t *testctx.T) {
+	t.Run("go sdk", func(ctx context.Context, t *testctx.T) {
+		testcases := []struct {
+			name          string
+			daggerjson    string
+			expectedValue string
+			expectedError string
+		}{
+			{
+				name: "goprivate is configured",
+				daggerjson: `{
+  "name": "foo",
+  "engineVersion": "v0.16.2",
+  "sdk": {
+    "source": "go",
+    "config": {
+      "goprivate": "github.com/foobar"
+    }
+  }
+}`,
+				expectedValue: "github.com/foobar",
+			},
+			{
+				name: "invalid type for goprivate is configured",
+				daggerjson: `{
+  "name": "foo",
+  "engineVersion": "v0.16.2",
+  "sdk": {
+    "source": "go",
+    "config": {
+      "goprivate": 1234
+    }
+  }
+}`,
+				expectedError: "'GoPrivate' expected type 'string', got unconvertible type 'float64', value: '1234'",
+			},
+		}
```

Comment:
Add another test for setting an unknown `foobar` settings, we should error on that as well.